code/game/machinery/iv_drip.dm

```diff
@@ -1,224 +1,700 @@
+				tank.forceMove(src)
+				tank_active = FALSE
+				valve_open = FALSE
+				update_icon()
+				return
+			if(tank.air_contents.return_pressure() == 0)
+				src.visible_message(SPAN_WARNING("\The [src] buzzes and automatically closes the valve."))
+				playsound(src, 'sound/machines/buzz-two.ogg', 50)
+				if(breather.internals)
+					breather.internals.icon_state = "internal0"
+				breather.internal = null
+				tank.forceMove(src)
+				tank_active = FALSE
+				valve_open = FALSE
+			if(epp) // Emergency Positive Pressure system forces respiration
+				if(breather.losebreath > 0)
+					if(!epp_active)
+						src.visible_message(SPAN_WARNING("\The [src] flashes a blue light, activating it's Emergency Positive Pressure system!"))
+						playsound(breather, 'sound/machines/windowdoor.ogg', 50)
+						epp_active = TRUE
+						update_icon()
+					tank.distribute_pressure = safe_pressure_min // Constantly adjusts the pressure to keep up with the damage
+					breather.losebreath = 0
+					to_chat(breather, SPAN_NOTICE("You feel fresh air being pushed into your lungs."))
+			update_icon()
+
+	if(istype(attached)) // IV attached
+		if(!beaker)
+			return
+		if(!attached.dna)
+			return
+		if(NOCLONE in attached.mutations)
+			return
+		if(attached.species.flags & NO_BLOOD)
+			return
+
+		if(!attached.Adjacent(src))
+			iv_rip()
+			update_icon()
+			return
 
+		if(mode) // Injecting
+			if(beaker.reagents.total_volume > 0)
+				beaker.reagents.trans_to_mob(attached, transfer_amount, CHEM_BLOOD)
+				update_icon()
+			if(toggle_stop) // Automatically detaches if the blood volume is at 100%
+				if((beaker.reagents.has_reagent(/decl/reagent/blood) || beaker.reagents.has_reagent(/decl/reagent/saline)) && attached.get_blood_volume() >= 100)
+					visible_message("\The <b>[src]</b> flashes a warning light, disengaging from [attached] automatically!")
+					playsound(src, 'sound/machines/buzz-two.ogg', 100)
+					attached = null
+					blood_message_sent = FALSE
+					update_icon()
+					return
+		else // Taking
+			var/amount = REAGENTS_FREE_SPACE(beaker.reagents)
+			amount = min(amount, transfer_amount)
+
+			if(amount == 0) // If the beaker is full, ping
+				if(prob(5))
+					visible_message("\The <b>[src]</b> pings.")
+					playsound(src, 'sound/machines/ping.ogg', 100)
+				return
+			if(attached.get_blood_volume() < 90 && !blood_message_sent)
+				visible_message(SPAN_WARNING("\The <b>[src]</b> flashes a warning light!"))
+				playsound(src, 'sound/machines/buzz-two.ogg', 100)
+				blood_message_sent = TRUE
+			if(attached.take_blood(beaker, amount))
+				update_icon()
+
+/obj/machinery/iv_drip/MouseDrop(over_object, src_location, over_location)
+	..()
 	if(in_range(src, usr) && ishuman(over_object) && get_dist(over_object, src) <= 1)
-		visible_message("[usr] attaches \the [src] to \the [over_object].")
-		src.attached = over_object
-		src.update_icon()
+		var/list/options = list(
+			"IV drip" = image('icons/mob/screen/radial.dmi', "iv_drip"),
```

Comment:
It is better to define things in this case, e.g.

`#define IV_DRIP "IV drip"`

So the compiler will yell at you if you make a spelling mistake, since you're typing this string out in many places.

Reply:
I've been using ``[src]`` more than ``IV drip`` whenever I can for this code to avoid typos involving the name. There's really only 4 instances of using ``IV drip``, one of them being the name, the desc_info, and the MouseDrop. I really don't think it's needed in this case

Reply:
There's no harm in using more defines than necessary, especially if you undef them after the codeblock. But this isn't critical, so I'll leave it up to you.

Reply:
Got it. I'll definitely keep it in mind for future code shenaniganry. Thank you!